src/battle_ai_util.c

```diff
@@ -372,9 +380,6 @@ s32 AI_CalcDamage(u32 move, u32 battlerAtk, u32 battlerDef, u8 *typeEffectivenes
     uq4_12_t effectivenessMultiplier;
     struct AiLogicData *aiData = AI_DATA;
 
-    SetBattlerData(battlerAtk);
-    SetBattlerData(battlerDef);
-
```

Comment:
If you remove `SetBattlerData` here wouldn't you need to add it before every other `AI_CalcDamage` call? (+ `RestoreBattlerData`).

Reply:
That's right. `AI_CalcDamageSaveBattlers` exists already which should be used during score calcs.

Though thinking about it now, maybe we can just call Save/SetbattlerData before score calcs and then we can remove all AI_xx equivalent functions

Reply:
> That's right. `AI_CalcDamageSaveBattlers` exists already which should be used during score calcs.

There are some `AI_CalcDamage` calls that don't go through `AI_CalcDamageSaveBattlers` so that's why I was curious.

Reply:
@ghoulslash could you respond to this? So we can get it merged.

Reply:
sorry for letting this slip. taking a closer look, I think you're right I need to handle the switch in check scenarios. I will need to think about if AI_CalcDamageSaveBattlers can just replace AI_CalcDamage in those scenarios, though

Reply:
okay, updated with upcoming and added `SetBattlerData` to AI_CalcPartyMonDamage for the opposing battler. This should be sufficient, since the switchInCandidate is already an AI battler so no need to guess at anything (and is also initialized in the switch AI code). So I actually think the previous setup was a bug, because Ai_CalcDamage would call SetBattlerData for the switchInCandidate and potentially overwrite the initialized data?
EIther way, ready for re-review